### ISSUE TITLE
Enforcing triple slash comments only on public APIs?

### ISSUE THREAD

**Issue by carlossanlop** [dotnet/roslyn#48868]
I added `<GenerateDocumentationFile>true</GenerateDocumentationFile>` to my csproj. This enables the rule CS1591 to get triggered whenever an API is not documented:

```
1>C:\Users\carlos\Source\Repos\testing\ConsoleCore\Library.cs(3,18,3,27): warning CS1591: Missing XML comment for publicly visible type or member 'MyLibrary'
```

The problem is that I only care about triple slash comments on top of public APIs. I don't care for private APIs. Is it possible to only have the warning show up for public APIs?

**Comment by Youssef1313** [comment]
The existing logic in Roslyn does seem to intend doing this.

https://github.com/dotnet/roslyn/blob/cf55f3a58e47298426fa971d3bd9d8857c746c65/src/Compilers/CSharp/Portable/Compiler/DocumentationCommentCompiler.cs#L549-L578

Probably a bug?

**Comment by jaredpar** [comment]
@carlossanlop can you provide a repro here? I did a quick test and this logic is behaving as I would expect here: do not warn on non-public members. 

```cs
// no warn here
class Program
{
    static void Main()
    {
    }
}

// warns here 
public class Results
{   

}
```

**Comment by carlossanlop** [comment]
@jaredpar 

MyNamespace.csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Library</OutputType>
    <TargetFramework>net5.0</TargetFramework>
    <GenerateDocumentationFile>true</GenerateDocumentationFile>
  </PropertyGroup>

</Project>
```

MyLibrary.cs
```cs
namespace MyNamespace
{
    public class MyLibrary
    {
        /// <summary>
        /// This is the public method summary.
        /// </summary>
        public void MyPublicMethod()
        {
        }

        private void MyPrivateMethod()
        {
        }
    }
}
```

The class gets the squiggle with CS1591:
![image](https://user-images.githubusercontent.com/1175054/97055679-27823d80-153c-11eb-97b4-c72dda4a59d4.png)

I am using VS Int Preview:
```
Microsoft Visual Studio Enterprise 2019 Int Preview
Version 16.9.0 Preview 1.0 [30612.141.main]
```

Which came with this version of 5.0 RC2 installed:
![image](https://user-images.githubusercontent.com/1175054/97055820-6d3f0600-153c-11eb-9ba7-e6521d126cf3.png)

**Comment by jaredpar** [comment]
@carlossanlop 

That behavior is expected though. The warning is that there is no doc comment on the type `MyLibrary`. That is a `public` type and hence needs a comment.

**Comment by carlossanlop** [comment]
Yes. I was just going to mention that this wasn't a bug. My bad. Thank you.
